### DISCUSSION TITLE
Ghost Queues in rabbitmq cluster

### DISCUSSION THREAD

**Discussion by anmol1vw13** [rabbitmq/rabbitmq-server#5386] [Other]
Hey folks,
There's a ghost queue in our Rabbitmq cluster. We use the below command to delete it but it re-appears after a while. And the consumer which is consuming from this queue gets blocked. I feel, there’s some reference that still lingers around even after the deletion.
Any suggestion on how to permanently delete it?
```
rabbitmqctl eval '{ok, Q} = rabbit_amqqueue:lookup(rabbit_misc:r(<<"/">>, queue, <<"queue-name">>)), rabbit_amqqueue:delete_crashed(Q).'
```

RabbitMQ Cluster version: 3.9.11

**Comment by michaelklishin** [comment]
We cannot suggest much with this amount of information.

No logs, details on queue type and settings, nothing to work with. On top of that, you are 10 patch releases behind the latest 3.9.

We used to see this with classic mirrored queues that fail to elect a leader either due to their [mirror promotion settings](https://rabbitmq.com/ha.html#unsynchronised-mirrors) or in scenarios that cannot be handled by the replication protocol of CMQs. They are deprecated and will be removed in RabbitMQ 4.0. Please switch to quorum queues for queues that must be replicated.

My best guess is that a classic mirrored queue ends up without a promotable leader and this has zero replicas running. Logs would make it easy to verify this hypothesis.
